### ISSUE TITLE
Invoke-WebRequest decodes URI-encoded URL string

### ISSUE THREAD

**Issue by dariok** [PowerShell/PowerShell#10871]
# Steps to reproduce

```powershell
PS /home/dario/diarium-processing> $uri1 = New-Object System.Uri -ArgumentList ("https://localhost/ingest/jg1798/07%2F1798-07-25.xml")
PS /home/dario/diarium-processing> $uri2 = New-Object System.Uri -ArgumentList ("https://localhost/ingest/jg1798/07%2F1798-07-25.xml", "true")
PS /home/dario/diarium-processing> $uri3 = "https://localhost/ingest/jg1798/07%2F1798-07-25.xml"

PS /home/dario/diarium-processing> Invoke-WebRequest $uri1 -Method Post -WebSession $diasession -ContentType application/xml -InFile $file
Invoke-WebRequest :                      
404 Not Found
Not Found
The requested URL /exist/restxq/edoc/admin/ingest/jg1798/07/1798-07-25.xml was not found on this server.

PS /home/dario/diarium-processing> $uri.UserEscaped
True
PS /home/dario/diarium-processing> Invoke-WebRequest $uri2 -Method Post -WebSession $diasession -ContentType application/xml -InFile $file
Invoke-WebRequest :                      
404 Not Found
Not Found
The requested URL /exist/restxq/edoc/admin/ingest/jg1798/07/1798-07-25.xml was not found on this server.

PS /home/dario/diarium-processing> Invoke-WebRequest $uri3 -Method Post -WebSession $diasession -ContentType application/xml -InFile $file
Invoke-WebRequest :                      
404 Not Found
Not Found
The requested URL /exist/restxq/edoc/admin/ingest/jg1798/07/1798-07-25.xml was not found on this server.

```

# Expected behavior
The URL used should be `https://localhost/ingest/jg1798/07%2F1798-07-25.xml`
A URL-encoded string should not be decoded before send. This is especially true if, as in `$uri2` above, `$uri2.UserEscaped` is set to `true` (or is this a complete misnoma?).

# Actual behavior
The URL used is `https://localhost/ingest/jg1798/07/1798-07-25.xml`, all URL-encoding is decoded and a wrong URL is used, rendering PowerShell unsuable for this request.

# Environment data
```powershell
PS /home/dario/diarium-processing> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.2.3
PSEdition                      Core
GitCommitId                    6.2.3
OS                             Linux 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

*Edit:*
# Status of possible workarounds
I consulted StackOverflow (https://stackoverflow.com/questions/43129163/powershell-invoke-webrequest-to-a-url-with-literal-2f-in-it and similar). The suggestions there do not work:

```powershell
PS /home/dario/diarium-processing>   $UnEscapeDotsAndSlashes = 0x2000000;
PS /home/dario/diarium-processing>   $SimpleUserSyntax = 0x20000;
PS /home/dario/diarium-processing> 
PS /home/dario/diarium-processing>   $type = $uri.GetType();
PS /home/dario/diarium-processing>   $fieldInfo = $type.GetField("m_Syntax", ([System.Reflection.BindingFlags]::Instance -bor [System.Reflection.BindingFlags]::NonPublic));
PS /home/dario/diarium-processing> 
PS /home/dario/diarium-processing>   $uriParser = $fieldInfo.GetValue($uri);
You cannot call a method on a null-valued expression.
At line:1 char:3
+   $uriParser = $fieldInfo.GetValue($uri);
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [], RuntimeException
+ FullyQualifiedErrorId : InvokeMethodOnNull

```

**Comment by iSazonov** [comment]
@dariok Thanks for your report! 

It seems it is .Net Core issue. For your example:
```powershell
 $uri2

AbsolutePath   : /ingest/jg1798/07%2F1798-07-25.xml
AbsoluteUri    : https://localhost/ingest/jg1798/07%2F1798-07-25.xml
LocalPath      : /ingest/jg1798/07/1798-07-25.xml
Authority      : localhost
```
Here LocalPath is already broken.
I suggest you open new issue in CoreFX repository. Better with C# repo example.

**Comment by mklement0** [comment]
I think there's a misconception here, and both PowerShell and CoreFX behave as designed:

When you construct a `System.URI` instance and ask it to escape the URI string you pass for you (the default behavior), it only escapes those parts that _aren't already escaped_:

```powershell
PS> [uri]::new('http://example.org/a%2Fb c').AbsoluteUri

http://example.org/a%2Fb%20c
```

Note how `%2F` was left alone, because it was already escaped, and only the space was encoded as `%20`.

---

By contrast, if you pass `$true` as the 2nd constructor argument, you indicate that the URI is _already pre-escaped:

```powershell
PS> [uri]::new('http://example.org/a%2Fb c', $true).AbsoluteUri

http://example.org/a%2Fb c
```

Note how the space char., even though it normally requires escaping as `%20`, was left as-is.

In this case it is your responsibility to provide  a properly _escaped_ URI.

(Given the desired-state logic of the _automatic_ escaping, I'm unclear on when this approach is needed / useful.)

---

**In both cases it is the string returned by `.AbsoluteUri` that will be used to make the request, and that string is expected to be _escaped_.**

* In your case, the `%2F` is therefore _decoded to `/`_, as expected.

* If you want to include `%2F` _verbatim_ in your URI, you need to encode _that_, and that means: `%252F`, which you can obtain programmatically with `[uri]::EscapeDataString('%2F')`; that is, the `%` character itself must be escaped, and given that its Unicode code point is `0x25`, an escaped `%` is `%25`

In short, use the following:

```powershell
# Note the '%25F' instead of '%2F'
$uri = New-Object System.Uri -ArgumentList "https://localhost/ingest/jg1798/07%25F1798-07-25.xml", $true
```

or, more succinctly (PSv5+):

```powershell
# Note the '%25F' instead of '%2F'
$uri = [uri]::new("https://localhost/ingest/jg1798/07%25F1798-07-25.xml", $true)
```

Note that the `$true` argument isn't strictly necessary in this case, however.

**Comment by iSazonov** [comment]
From .Net team comment it is the server error, not client or .Net.
